specification/network/resource-manager/Microsoft.Network/stable/2020-08-01/virtualNetworkGateway.json

```diff
@@ -3212,6 +3269,156 @@
+      },
+      "integrity": {
+        "type": "string",
+        "$ref": "#/definitions/IpsecIntegrity",
+        "description": "The IPSec integrity algorithm (IKE phase 2)."
+      },
+      "pfsGroupId": {
+        "type": "string",
+        "$ref": "#/definitions/PfsGroup",
+        "description": "The PFS Group"
+      },
+      "inboundSPI": {
+        "type": "integer",
+        "format": "int32",
+        "description": "The Inbound SPI SA"
+      },
+      "outboundSPI": {
+        "type": "integer",
+        "format": "int32",
+        "description": "The Outbound SPI SA"
+      },
+      "localTrafficSelectors": {
+        "type": "array",
+        "items": {
+          "type": "string"
+        },
+        "description": "The local traffic selectors."
+      },
+      "remoteTrafficSelectors": {
+        "type": "array",
+        "items": {
+          "type": "string"
+        },
+        "description": "The remote traffic selectors."
+      },
+      "lifetimeKilobytes": {
+        "type": "integer",
+        "format": "int64",
+        "minimum": 1024,
+        "maximum": 102400000,
+        "description": "The QM SA Lifetime KBytes."
+      },
+      "lifeTimeSeconds": {
+        "type": "integer",
+        "format": "int64",
+        "minimum": 300,
+        "maximum": 27000,
+        "description": "The QM SA Lifetime Seconds."
+      },
+      "isSaInitiator": {
+        "type": "boolean",
+        "description": "Flag to check if SA Initiator or not."
+      },
+      "elapsedTimeInseconds": {
+        "type": "integer",
+        "format": "int32",
+        "description": "The time elapsed in seconds."
+      }
+    },
+    "MainModeSaPropertiesFormat": {
+      "localEndpoint": {
+        "type": "string",
+        "description": "The local endpoint."
+      },
+      "remoteEndpoint": {
+        "type": "string",
+        "description": "The remote endpoint."
+      },
+      "initiatorCookie": {
+        "type": "integer",
+        "format": "int64",
+        "description": "The initiator cookie."
+      },
+      "responderCookie": {
+        "type": "integer",
+        "format": "int64",
+        "description": "The responder cookie."
+      },
+      "localUdpEncapsulationPort": {
+        "type": "integer",
+        "format": "int32",
+        "description": "The local UDP encapsulation port."
+      },
+      "remoteUdpEncapsulationPort": {
+        "type": "integer",
+        "format": "int32",
+        "description": "The remote UDP encapsulation port."
+      },
+      "encryption": {
+        "type": "string",
+        "$ref": "#/definitions/IkeEncryption",
+        "description": "The IKE encryption algorithm (IKE phase 1)."
+      },
+      "integrity": {
+        "type": "string",
+        "$ref": "#/definitions/IkeIntegrity",
+        "description": "The IKE integrity algorithm (IKE phase 1)."
+      },
+      "dhGroup": {
+        "type": "string",
+        "$ref": "#/definitions/DhGroup",
```

Comment:
these enums are predefined are you sure the ordering etc will not make an issue. if not it would be better as string?

Reply:
Changed to string